$Id: fix_graphviz.patch 344 2012-06-29 15:38:56Z tg $

Fix:
* path to dot
* XHTML/1.0 Transitional validity of output

--- a/dist/mediawiki-extensions-graphviz/usr/share/mediawiki-extensions/graphviz/GraphViz.php
+++ b/dist/mediawiki-extensions-graphviz/usr/share/mediawiki-extensions/graphviz/GraphViz.php
@@ -17,7 +17,7 @@
  *	Describes where your actual (dot) executable remains.
  *
  *	Windows Default: C:/Programme/ATT/Graphviz/bin/
- *	Other Platform : /usr/local/bin/dot
+ *	Other Platform : /usr/bin/dot
  *
  * $wgGraphVizSettings->mscgenPath
  *			Describes where your actual mscgen-executable remains
@@ -440,9 +440,10 @@ $wgExtensionCredits['parserhook'][] = ar
 			{
 				$txt = imageAtrributes( $args, $storagename, $map, $outputType, $wgUploadPath ); // if we want borders/position/...
 			} else {
-				$txt  = '<map name="' . $storagename . '">' . $map . '</map>' .
-					 '<img src="' . $wgUploadPath . '/graphviz/' . $storagename . '.' . $outputType . '"' .
-							   ' usemap="#' . $storagename . '" />';
+				$txt  = '<map id="map-' . md5($storagename) . '" name="' . $storagename . '"><p>' . $map . '</p></map>' .
+					'<img alt="map-' . $storagename . '"' .
+					' src="' . $wgUploadPath . '/graphviz/' . $storagename . '.' . $outputType . '"' .
+					' usemap="#' . $storagename . '" />';
 			}
 
 
@@ -613,10 +614,10 @@ $wgExtensionCredits['parserhook'][] = ar
 		}
 
 		// Produce the basic html
-		$txt  = '<map name="' . $storagename . '">' . $map . '</map>' .
-					 '<img class="' . $imageClass . '" style="' . $imageStyle . '"' .
-					  'alt="' . $alt . '" src="' . $wgUploadPath . '/graphviz/' . $storagename . '.' . $outputType . '"' .
-							   ' usemap="#' . $storagename . '" />';
+		$txt  = '<map id="map-' . md5($storagename) . '" name="' . $storagename . '"><p>' . $map . '</p></map>' .
+			'<img class="' . $imageClass . '" style="' . $imageStyle . '"' .
+			' alt="' . $alt . '" src="' . $wgUploadPath . '/graphviz/' . $storagename . '.' . $outputType . '"' .
+			' usemap="#' . $storagename . '" />';
 
 		// Add necessary containers
 		if ( $caption != null ) {
